From 7f462228f179c04ad7dfd1a43de39feaf8dfb607 Mon Sep 17 00:00:00 2001
From: Jeffrey Stedfast <fejj@gnome.org>
Date: Sun, 29 Dec 2013 22:38:13 +0000
Subject: Fixed a memory leak in g_mime_object_set_content_disposition_parameter

2013-12-29  Jeffrey Stedfast  <fejj@gnome.org>

	* gmime/gmime-object.c
	(g_mime_object_set_content_disposition_parameter): Unref the
	disposition. Thanks to Mario Theodoridis for this patch.
---
diff --git a/gmime/gmime-object.c b/gmime/gmime-object.c
index 6d326f2..790e41e 100644
--- a/gmime/gmime-object.c
+++ b/gmime/gmime-object.c
@@ -647,6 +647,7 @@ g_mime_object_set_content_disposition_parameter (GMimeObject *object, const char
 	if (!object->disposition) {
 		disposition = g_mime_content_disposition_new ();
 		_g_mime_object_set_content_disposition (object, disposition);
+		g_object_unref (disposition);
 	}
 	
 	g_mime_content_disposition_set_parameter (object->disposition, name, value);
--
cgit v0.9.2
